scripts/test_solidity_lsp.py

```diff
@@ -0,0 +1,379 @@
+
+    def call_method(self, method_name: str, params: Any) -> Any:
+        self.send_message(method_name, params)
+        return self.receive_message()
+
+    def send_notification(self, name: str, params: Any = None) -> None:
+        self.send_message(name, params)
+
+# }}}
+
+SGR_RESET = '\033[m'
+SGR_TRACE = '\033[1;36m'
+SGR_TEST_BEGIN = '\033[1;33m'
+SGR_ASSERT_BEGIN = '\033[1;34m'
+SGR_STATUS_OKAY = '\033[1;32m'
+SGR_STATUS_FAIL = '\033[1;31m'
+
+class ExpectationFailed(Exception):
+    def __init__(self, actual, expected):
+        self.actual = actual
+        self.expected = expected
+        diff = DeepDiff(actual, expected)
+        super().__init__(
+            f"Expectation failed. Expected {expected} but got {actual}. {diff}"
+        )
+
+# pylint: disable-next=too-many-instance-attributes
+class SolidityLSPTestSuite: # {{{
+    tests_passed: int = 0
+    tests_failed: int = 0
+    assertion_count: int = 0   # number of total assertions executed so far
+    assertions_passed: int = 0
+    assertions_failed: int = 0
+    print_assertions: bool = False
+
+    def __init__(self):
+        self.solc_path, self.project_root_dir, self.print_assertions = self.parse_args_and_prepare()
+        self.project_root_uri = 'file://' + self.project_root_dir
+
+    def parse_args_and_prepare(self):
+        """
+        Parses CLI args and returns tuple of path to solc executable
+        and path to solidity-project root dir.
+        """
+        parser = argparse.ArgumentParser(description='Solidity LSP Test suite')
+        parser.set_defaults(print_assertions=False)
+        parser.add_argument(
+            '-v, --print-assertions',
+            dest='print_assertions',
+            action='store_true',
+            help='Be more verbose by also printing assertions.'
+        )
+        parser.add_argument(
+            'solc_path',
+            type=str,
+            default="/home/trapni/work/solidity/build/solc/solc",
+            help='Path to solc binary to test against',
+            nargs="?"
+        )
+        parser.add_argument(
+            'project_root_dir',
+            type=str,
+            default=f"{os.path.dirname(os.path.realpath(__file__))}/..",
+            help='Path to Solidity project\'s root directory (must be fully qualified).',
+            nargs="?"
+        )
+        args = parser.parse_args()
+        project_root_dir = os.path.realpath(args.project_root_dir) + '/test/libsolidity/lsp'
+        return [args.solc_path, project_root_dir, args.print_assertions]
+
+    def main(self) -> int:
+        """
+        Runs all test cases.
+        Returns 0 on success and the number of failing assertions (capped to 127) otherwise.
+        """
+
+        for method_name in sorted([name for name
+                            in dir(SolidityLSPTestSuite)
+                            if callable(getattr(SolidityLSPTestSuite, name)) and
+                                name.startswith("test_")]):
```

Comment:
I think it's more readable wrapped this way. And closer to how we wrap stuff in C++ - code is not shifted to the right so much that it forces you to wrap the `if` in an unnatural way:
```suggestion
        for method_name in sorted([
            name
            for name in dir(SolidityLSPTestSuite)
            if callable(getattr(SolidityLSPTestSuite, name)) and name.startswith("test_")
        ]):
```